engine/src/main/java/graphql/nadel/engine/execution/Execution.java

```diff
@@ -163,45 +169,82 @@ private static NadelInstrumentationExecuteOperationParameters buildInstrumentati
-                        fieldInfoByDefinition.put(graphQLFieldDefinition, fieldInfo);
-                    }
+            for (ObjectTypeDefinition subscriptionTypeDefinition : subscriptionTypeDefinitions) {
+                GraphQLObjectType schemaSubscriptionType = overallSchema.getSubscriptionType();
+                for (FieldDefinition fieldDefinition : subscriptionTypeDefinition.getFieldDefinitions()) {
+                    GraphQLFieldDefinition graphQLFieldDefinition = schemaSubscriptionType.getFieldDefinition(fieldDefinition.getName());
+                    FieldInfo fieldInfo = new FieldInfo(FieldInfo.FieldKind.TOPLEVEL, service, graphQLFieldDefinition);
+                    fieldInfoByDefinition.put(graphQLFieldDefinition, fieldInfo);
                 }
             }
         }
         return new FieldInfos(fieldInfoByDefinition);
     }
 
+    private void extractNamespacedFieldInfos(Map<GraphQLFieldDefinition, FieldInfo> fieldInfoByDefinition, Set<GraphQLOutputType> namespacedGraphqlTypes) {
+        for (Service service : services) {
+            for (SDLDefinition<?> definition : service.getDefinitionRegistry().getDefinitions()) {
+                if (!(definition instanceof ObjectTypeDefinition)) {
+                    continue;
+                }
+                ObjectTypeDefinition typeDefinition = (ObjectTypeDefinition) definition;
+                boolean isNamespacedDefinition = namespacedGraphqlTypes.stream()
+                        .filter(type -> type instanceof GraphQLNamedType)
+                        .map(type -> (GraphQLNamedType) type)
+                        .anyMatch(type -> type.getName().equals(typeDefinition.getName()));
+                if (!isNamespacedDefinition) {
+                    continue;
+                }
+                GraphQLObjectType namespacedGraphQLObjectType = overallSchema.getObjectType(typeDefinition.getName());
+                List<GraphQLFieldDefinition> graphQLFieldDefinitionsWithinNamespacedType = namespacedGraphQLObjectType.getFieldDefinitions();
+
+                List<FieldDefinition> serviceFieldDefinitionsWithinNamespacedType = typeDefinition.getFieldDefinitions();
+
+                for (FieldDefinition serviceSecondLevelFieldDefinition : serviceFieldDefinitionsWithinNamespacedType) {
+                    GraphQLFieldDefinition secondLeveGraphqlFieldDefinition = FpKit.findOne(
+                            graphQLFieldDefinitionsWithinNamespacedType,
+                            gqlDef -> gqlDef.getName().equals(serviceSecondLevelFieldDefinition.getName())
+                    ).get();
+
+                    FieldInfo fieldInfo = new FieldInfo(FieldInfo.FieldKind.NAMESPACE_SUBFIELD, service, secondLeveGraphqlFieldDefinition);
+                    fieldInfoByDefinition.put(secondLeveGraphqlFieldDefinition, fieldInfo);
+                }
+            }
+        }
+    }
+
```

Comment:
Must of the logic here is the same

Rather than retain the 2 code paths (name spaced and not name spaced)

Lets move this into `FieldInfos` and one "extract" method